### PR TITLE
AirfRANS: WD=0 ablation on vol-10x + EMA champion

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The DM champion uses **no weight decay** (WD=0) and achieves 3.833%. The AF champion uses WD=1e-2. Early AF experiments showed WD=1e-2 helped (PR #2709, pre-EMA), but that was in a fundamentally different training regime — no EMA, no vol-weight=10x, different architecture (4L/256d vs 2L/256d), and shorter training budgets.

**Why WD=0 might help now:**
1. **EMA provides implicit regularization.** EMA=0.999 already acts as a low-pass filter on the weight trajectory, providing the stability benefit that WD was originally needed for. With EMA, WD may be over-regularizing.
2. **Volume predictions may suffer from WD.** WD penalizes large weights uniformly. If volume predictions require larger weights to capture the wider dynamic range of volume fields (pressure varies more in the volume than at the surface), WD could be actively suppressing volume capacity.
3. **DM precedent.** DM with EMA=0.9995 and no WD works perfectly. AF with EMA=0.999 may similarly benefit.

This has never been tested with the current AF champion config (2L/256d, T_max=50, gc=1.0, EMA=0.999, vol-10x).

## Instructions

Run 2 trials. The only change from baseline is removing weight decay.

**Trial 1 — WD=0 (remove weight-decay entirely):**
```
cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 \
  --grad-clip 1.0 \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --vol-loss-weight 10.0 \
  --wandb_group af-no-wd
```
Note: explicitly do NOT pass `--weight-decay`.

**Trial 2 — WD=5e-3 (lighter WD, halfway between 0 and 1e-2):**
```
cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 \
  --grad-clip 1.0 --weight-decay 5e-3 \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --vol-loss-weight 10.0 \
  --wandb_group af-no-wd
```

Run each trial until timeout. Report val surface_mse and vol_mse at the best validation checkpoint.

## Baseline

Current AF best (PR #3135):
- **val surface_mse:** 0.000296 at epoch 704
- **val vol_mse:** 0.002039 at epoch 743
- **Goal:** vol_mse < 0.0017 without regressing surface_mse above 0.000296
- **W&B run:** sh2zyfwr
- **Reproduce:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 --ema-decay 0.999 --vol-loss-weight 10.0`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT rei:

## Results

### Trial 1 — WD=0 (run `sqotwbxs`, 771 epochs on GPU 0)

| Metric | Baseline (WD=1e-2) | WD=0 | Δ |
|--------|-------------------|------|---|
| **val surface_mse** | 0.000296 (ep704) | 0.000384 (ep636) | **+29.7% (REGRESSED)** |
| **val vol_mse** (best across all epochs) | 0.002039 (ep743) | 0.002474 (ep753) | **+21.3% (REGRESSED)** |
| test surface_mse | — | 0.000892 | — |
| test vol_mse | — | 0.003793 | — |

**Command:**
```
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 0 \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 \
  --agent rei --wandb-name "rei/af-wd0" --wandb-group af-no-wd
```

### Trial 2 — WD=5e-3 (run `cjcftqo8`, 764 epochs on GPU 1)

| Metric | Baseline (WD=1e-2) | WD=5e-3 | Δ |
|--------|-------------------|---------|---|
| **val surface_mse** | 0.000296 (ep704) | **0.000249 (ep682)** | **-15.9% (NEW BEST!)** |
| **val vol_mse** (best across all epochs) | 0.002039 (ep743) | 0.002414 (ep692) | **+18.4% (REGRESSED)** |
| test surface_mse | — | 0.000833 | — |
| test vol_mse | — | 0.003468 | — |

**Command:**
```
cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 5e-3 \
  --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --volume-loss-weight 10.0 \
  --agent rei --wandb-name "rei/af-wd5e3" --wandb-group af-no-wd
```

### Peak memory
~16.2 GB VRAM per trial (2L/256d AirfRANS)

### What happened

**WD=0 is clearly worse on both metrics.** The hypothesis that EMA=0.999 provides sufficient implicit regularization to replace weight decay is **not supported** for AirfRANS. Without WD, both surface_mse (+29.7%) and vol_mse (+21.3%) regress substantially. The DM precedent (WD=0 works with EMA) does not transfer to AF — likely because AF's 2L/256d architecture with AdamW has different optimization dynamics than DM's 4L/512d.

**WD=5e-3 improves surface_mse by 15.9% (new programme best!) but hurts vol_mse by 18.4%.** Lighter WD allows the model to fit surface details more precisely, but this comes at the cost of volume prediction capacity. The surface/volume tradeoff suggests that WD=1e-2 was actually providing useful regularization for the volume predictions (which have larger dynamic range and benefit from weight constraints), while slightly over-regularizing the surface predictions.

**The vol_mse goal of <0.0017 was not achieved** by either trial. Both are worse than the baseline's 0.002039.

### Suggested follow-ups

1. **WD=5e-3 + higher vol-loss-weight (15x or 20x):** Since WD=5e-3 unlocks surface gains but hurts volume, compensating with more volume emphasis could recover both wins simultaneously.
2. **Differential WD per parameter group:** Surface-head parameters at WD=5e-3, volume-head parameters at WD=1e-2. This could capture the surface improvement without sacrificing volume.
3. **WD=7e-3 (between 5e-3 and 1e-2):** A tighter bracket around the optimal WD for balanced surface/volume performance.

**Comment by tcapelle** [comment]
ADVISOR: Excellent finding — WD=5e-3 surface_mse=0.000249 is a new programme best (-15.9%), confirming that lighter WD unlocks surface precision. However, vol_mse=0.002679 at the best-surface checkpoint (+31.4%) is too far from our target of <0.0017 to merge.

The surface/volume tradeoff you discovered is the most interesting AF result in several cycles. The natural follow-up is to compensate by increasing vol-loss-weight while keeping WD=5e-3:

**Trial 1 — WD=5e-3 + vol-loss-weight=15x:**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 \
  --weight-decay 5e-3 --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 \
  --vol-loss-weight 15.0 \
  --wandb_group af-wd5e3-vol-weight-sweep
```

**Trial 2 — WD=5e-3 + vol-loss-weight=20x:**
```bash
cd target/icml2026 && python train.py \
  --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 \
  --weight-decay 5e-3 --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 \
  --vol-loss-weight 20.0 \
  --wandb_group af-wd5e3-vol-weight-sweep
```

The question: can vol-weight=15x/20x recover vol_mse toward baseline's 0.002039 while keeping surface near your 0.000249? If yes, that's a simultaneous improvement on both metrics.

**Comment by tcapelle** [comment]
STUDENT rei:

## Results — Follow-up: WD=5e-3 + vol-weight sweep

### Trial 1 — WD=5e-3 + vol-loss-weight=15x (run `nfqb7ubi`, GPU 0)

**DIVERGED** — training diverged after epoch ~299. Grad norms went to Infinity, final loss=19.0.

| Metric | Baseline (WD=1e-2, vol=10x) | WD=5e-3, vol=10x (prev) | WD=5e-3, vol=15x | Δ vs baseline |
|--------|-------------------|-------------------------|-------------------|---------------|
| **val surface_mse** | 0.000296 (ep704) | 0.000249 (ep682) | 0.000800 (ep299) | **+170% (DIVERGED)** |
| **val vol_mse** | 0.002039 (ep743) | 0.002414 (ep692) | 0.005862 (ep299) | **+187% (DIVERGED)** |
| test surface_mse | — | 0.000833 | 0.001947 | — |
| test vol_mse | — | 0.003468 | 0.006981 | — |

**Command:**
```
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 \
  --weight-decay 5e-3 --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --volume-loss-weight 15.0 \
  --agent rei --wandb-name "rei/af-wd5e3-vol15x" --wandb-group af-wd5e3-vol-weight-sweep
```

### Trial 2 — WD=5e-3 + vol-loss-weight=20x (run `vxe7c5dz`, GPU 1)

Trained stably to epoch 764.

| Metric | Baseline (WD=1e-2, vol=10x) | WD=5e-3, vol=10x (prev) | WD=5e-3, vol=20x | Δ vs baseline |
|--------|-------------------|-------------------------|-------------------|---------------|
| **val surface_mse** | 0.000296 (ep704) | 0.000249 (ep682) | 0.000382 (ep764) | **+29.1% (REGRESSED)** |
| **val vol_mse** | 0.002039 (ep743) | 0.002414 (ep692) | 0.002467 (ep764) | **+21.0% (REGRESSED)** |
| test surface_mse | — | 0.000833 | 0.000580 | — |
| test vol_mse | — | 0.003468 | 0.003001 | — |

**Command:**
```
cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 \
  --weight-decay 5e-3 --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 \
  --epochs 999 --ema-decay 0.999 --volume-loss-weight 20.0 \
  --agent rei --wandb-name "rei/af-wd5e3-vol20x" --wandb-group af-wd5e3-vol-weight-sweep
```

### Peak memory
~16.2 GB VRAM per trial

### What happened

**Trial 1 (vol-15x) diverged completely.** Grad norms hit Infinity sometime after epoch 299, and the model never recovered. The EMA checkpoint at epoch 299 had reasonable but poor metrics (surface_mse=0.000800). The divergence may be a stability issue where vol-weight=15x creates an unstable loss landscape with WD=5e-3.

**Trial 2 (vol-20x) trained stably but failed on both metrics.** Increasing vol-weight from 10x to 20x:
- **Hurt surface_mse** (0.000382 vs 0.000249 at vol-10x) — as expected, less relative surface weight
- **Did not help vol_mse** (0.002467 vs 0.002414 at vol-10x) — the volume metric is basically unchanged despite doubling the volume loss emphasis

This is the key finding: **vol_mse is insensitive to vol-weight changes under WD=5e-3**. The model's volume capacity limitation comes from the WD regime itself, not the loss weighting. 2x more vol-weight buys 0% vol improvement while costing 53% in surface quality.

**The vol_mse target of <0.0017 was not achieved** by any WD=5e-3 configuration, regardless of vol-weight (10x/15x/20x).

### Summary across all 4 trials in this PR

| Config | val surface_mse | Δ vs BL | val vol_mse | Δ vs BL | Stable? |
|--------|----------------|---------|-------------|---------|---------|
| Baseline (WD=1e-2, vol=10x) | 0.000296 | — | 0.002039 | — | ✅ |
| WD=0, vol=10x | 0.000384 | +29.7% | 0.002474 | +21.3% | ✅ |
| WD=5e-3, vol=10x | **0.000249** | **-15.9%** | 0.002414 | +18.4% | ✅ |
| WD=5e-3, vol=15x | 0.000800 | +170% | 0.005862 | +187% | ❌ diverged |
| WD=5e-3, vol=20x | 0.000382 | +29.1% | 0.002467 | +21.0% | ✅ |

**Conclusion:** WD=5e-3 unlocks a genuine surface improvement (-15.9%) but the volume regression (~+20%) cannot be compensated by increasing vol-weight alone. The WD=1e-2 baseline remains the best balanced configuration for AF. The surface improvement from WD=5e-3 is real but comes at an unrecoverable volume cost.

### Suggested follow-ups

1. **Differential WD per parameter group:** Surface-head at WD=5e-3, encoder+volume-head at WD=1e-2. This is the most targeted way to capture the surface gain without affecting volume regularization.
2. **WD=7e-3 (split the bracket):** A tighter search between 5e-3 and 1e-2 might find a balanced point — though the insensitivity of vol_mse to vol-weight suggests the issue is structural.
3. **Close this line of inquiry:** WD=1e-2 remains optimal for balanced AF performance. The surface improvement from WD=5e-3 is interesting but not actionable without solving the volume regression.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3238 because ADVISOR: Closing — WD=5e-3 gives programme-best surface (0.000249) but vol_mse=0.002414 can't be brought below 0.002039 ceiling across 3 follow-up attempts (vol=15x diverged, vol=20x gives 0.002467). Surface/volume tradeoff avenue exhausted. Rei reassigned..
